downloads/swan-lake-release-notes/swan-lake-2201.9.0/RELEASE_NOTE.md

```diff
@@ -68,6 +82,12 @@ To view bug fixes, see the [GitHub milestone for Swan Lake Update 9 (2201.9.0)](
 
 #### Language Server
 
+- Improved the snippet completion items provided for dependently typed functions.
+- Improved the completion items provided for precise resource parameter types.
```

Comment:
What do we mean by " precise resource parameter types"?

Reply:
The change focuses on the precise type described in the [spec](https://ballerina.io/spec/lang/master/#section_6.2.3:~:text=The%20most%20important,is%20string). However, I rewrote it using singleton types presuming it would be clearer.